src/renderer/components/outputs/OutputContainer.tsx

```diff
@@ -1,63 +1,172 @@
+    ({
+        children,
+        hasHandle,
+        outputId,
+        id,
+        type,
+    }: React.PropsWithChildren<OutputContainerProps>) => {
+        const { isValidConnection, edgeChanges, useConnectingFromType, useConnectingFrom } =
+            useContext(GlobalVolatileContext);
+        const { useIsDarkMode } = useContext(SettingsContext);
+        const [isDarkMode] = useIsDarkMode;
+
+        const { getEdges } = useReactFlow();
+        const edges = useMemo(() => getEdges(), [edgeChanges]);
+        const isConnected = !!edges.find(
+            (e) => e.source === id && parseHandle(e.sourceHandle!).inOutId === outputId
+        );
+        const [connectingFromType] = useConnectingFromType;
+        const [connectingFrom] = useConnectingFrom;
+
+        const showHandle = useMemo(() => {
+            if (
+                !connectingFrom ||
+                !connectingFromType ||
+                // We want to display the connectingFrom handle
+                (connectingFrom.handleId === `${id}-${outputId}` &&
+                    connectingFrom.handleType === 'source')
+            ) {
+                return true;
+            }
+            if (connectingFrom.nodeId === id) {
+                return false;
+            }
+            if (connectingFrom.handleType === 'source') {
+                return false;
+            }
+            const connectionIsValid = isValidConnection({
+                source: connectingFrom.nodeId,
+                sourceHandle: connectingFrom.handleId,
+                target: id,
+                targetHandle: `${id}-${outputId}`,
+            });
+            if (connectionIsValid && intersect(connectingFromType, type).type !== 'never') {
+                return true;
+            }
+            return false;
+        }, [connectingFromType, connectingFrom, type, id, outputId]);
+
+        const { typeDefinitions } = useContext(GlobalContext);
 
         let contents = children;
-        const handleColor = useColorModeValue('#EDF2F7', '#171923');
-        const borderColor = useColorModeValue('#171923', '#F7FAFC');
+        const [handleColor] = getTypeAccentColors(type, typeDefinitions, isDarkMode); // useColorModeValue('#EDF2F7', '#171923');
+        const connectedColor = useColorModeValue('#EDF2F7', '#171923');
         if (hasHandle) {
             contents = (
                 <HStack
                     h="full"
                     sx={{
                         '.react-flow__handle-connecting': {
-                            background: '#E53E3E !important',
-                            // cursor: 'not-allowed !important',
+                            // background: '#E53E3E !important',
+                            opacity: showHandle ? 1 : 0,
                         },
                         '.react-flow__handle-valid': {
-                            background: '#38A169 !important',
+                            // background: '#38A169 !important',
```

Comment:
Why is there so much commented out code everywhere?

Reply:
Because I forgot to remove it all